app/models/plage_ouverture.rb

```diff
@@ -47,10 +51,20 @@ def self.for_motif_and_lieu_from_date_range(motif_name, lieu, inclusive_date_ran
     results.uniq
   end
 
+  def expired?
+    # Use .expired_cached? for performance
+    (recurrence.nil? && first_day < Date.today) ||
+      (recurrence.present? && recurrence.to_hash[:until].present? && recurrence.to_hash[:until].to_date < Date.today)
+  end
+
   def available_motifs
     Motif.available_motifs_for_organisation_and_agent(organisation, agent)
   end
 
+  def refresh_plage_ouverture_expired_cached
+    update_column(:expired_cached, expired?)
```

Comment:
i'm unsure this is a good idea to use `update_column` instead of `update!`.
with update_column : 


> Validations are skipped.
> Callbacks are skipped.
> updated_at/updated_on are not updated.

Reply:
so after discussion, good call as we **need** to skip callbacks here to avoid infinite loop 👍